scripts/record_tennicam_vicon_trajectory.py

```diff
@@ -0,0 +1,150 @@
+import tennicam_client
+from vicon_transformer import pam_vicon_o80
+
+
+class JsonEncoder(json.JSONEncoder):
+    """JSON encoder that handles custom types used here"""
+
+    def default(self, obj: t.Any) -> t.Any:
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self, obj)
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "destination",
+        type=pathlib.Path,
+        help="File to which the recorded data is written.",
+    )
+    parser.add_argument(
+        "--rate",
+        "-r",
+        type=float,
+        default="1.0",
+        metavar="SECONDS",
+        help="Rate in seconds at which positions are saved.  Default: %(default)s",
+    )
+    parser.add_argument(
+        "--tennicam-segment-id",
+        type=str,
+        default="tennicam_client",
+        help="""Shared memory segment ID used by the tennicam back end.
+            Default: "%(default)s"
+        """,
+    )
+    parser.add_argument(
+        "--vicon-segment-id",
+        type=str,
+        default="vicon",
+        help="""Shared memory segment ID used by the vicon back end.
+            Default: "%(default)s"
+        """,
+    )
+    parser.add_argument(
+        "--verbose", "-v", action="store_true", help="Enable verbose output."
+    )
+    args = parser.parse_args()
+
+    logging.basicConfig(
+        level=logging.DEBUG if args.verbose else logging.INFO,
+        format="[%(asctime)s] [%(name)s | %(levelname)s] %(message)s",
+    )
+
+    if args.destination.exists():
+        logging.fatal("Destination file %s already exists.", args.destination)
+        return 1
+
+    logging.debug("Tennicam segment id: %s", args.tennicam_segment_id)
+    logging.debug("Vicon segment id: %s", args.vicon_segment_id)
+    logging.debug("Recording rate: %f seconds", args.rate)
+
+    tennicam_frontend = tennicam_client.FrontEnd(args.tennicam_segment_id)
+    vicon_frontend = pam_vicon_o80.FrontEnd(args.vicon_segment_id)
+
+    data = []
+    signal_handler.init()  # for detecting ctrl+c
+    with contextlib.suppress(KeyboardInterrupt, SystemExit), tqdm.tqdm() as progress:
+        next_update = time.time() + args.rate
+        while not signal_handler.has_received_sigint():
+            now = time.time()
+            sleep_duration = max(0, next_update - now)
+            time.sleep(sleep_duration)
+            next_update += args.rate
+
+            # Tennicam is a bit slower than Vicon (~180 Hz vs 300 Hz).  Therefore, wait
+            # for the next tennicam observation and then immediately get the latest
+            # Vicon observation.  This way, they should be fairly synchronised.
+            logging.debug("tennicam: Wait for next observation.")
+            tennicam_frontend.reset_next_index()
+            obs_tennicam = tennicam_frontend.wait_for_next()
```

Comment:
@vincentberenz Am I using this correctly here?  My goal is to ignore all existing observations and wait for the next one to arrive (i.e. current index + 1), so that when I get the observation it is as new as possible.
Then below, I use `vicon_frontend.latest()` to get the Vicon observation that is closest to the tennicam observation I just got.

Reply:
'reset_next_index' should be called only once (line 101).

Reply:
I tried this at first, but it resulted in giving me all observations one by one.  The data is recorded with a reduced rate (something like one sample per second) and my goal is to skip all observations between two iterations and always wait until a new one arrives (to make sure there is as little delay as possible).

So basically I could do
```python
i = frontend.latest().get_iteration()
obs = frontend.read(i+1)
```
inside the loop but was thinking that `wait_for_next()` results in nicer looking code (assuming it does the same thing).

Reply:
oh, sorry, I misunderstood the intent. So all good. I would have a preference for the second alternative (```read(i+1)```)

Reply:
Okay, I changed it accordingly.  I'll merge once it's tested.

Reply:
I fixed a mistake and also needed to add a try/catch-block around the `read()` call.  The problem is that pressing Ctrl+C while `read()` is currently waiting results in an exception (I think it's coming from inside the time series implementation to unblock the application in that case).
This is now tested and working.